### PR TITLE
Persist Average Daily Dose input between data views

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
See https://trello.com/c/8w4zfRPq for details.

Goes with tidepool-org/blip#542

### CODE REVIEW

**Comment by cbwebdevelopment** [comment]
[Regression: User-entered data for the new Avg. Daily Insulin widget does not persist](https://trello.com/c/8w4zfRPq/578-regression-user-entered-data-for-the-new-avg-daily-insulin-widget-does-not-persist)

File: test/components/common/stat/Stat.test.js [hunk 1]
```diff
@@ -570,6 +570,14 @@ describe('Stat', () => {
       expect(inputGroup().props().onChange).to.be.a.function;
       expect(inputGroup().props().onSuffixChange).to.be.a.function;
     });
+
+    it('should should call the `onInputChange` Stat prop when provided on input change', () => {
+
```
Comment (krystophv) [inline test/components/common/stat/Stat.test.js]: Bit of an echo in here. I think these tests might have been glanced over after being stubbed.

File: test/components/common/stat/Stat.test.js [hunk 2]
```diff
@@ -2564,5 +2593,40 @@ describe('Stat', () => {
 
       expect(wrapper.state().inputSuffix.value).to.equal(600);
     });
+
+    it('should call `propagateInputChange` after setting state', () => {
```
Comment (krystophv) [inline test/components/common/stat/Stat.test.js]: This is the same description as one added just above. Both are testing various state setting, but maybe the descriptions could be differentiated just a little?

File: src/components/common/stat/Stat.js [hunk 3]
```diff
@@ -886,15 +887,21 @@ class Stat extends PureComponent {
...
     this.setState((state) => ({
       inputSuffix: _.assign({}, state.inputSuffix, {
         value,
       }),
-    }));
+    }), this.propagateInputChange);
+  };
+
+  propagateInputChange = () => {
+    if (typeof this.props.onInputChange === 'function') {
```
Comment (krystophv) [inline src/components/common/stat/Stat.js]: Not a blocker, but should we consider using `_.isFunction` here as it's mildly more robust?

Comment (cbwebdevelopment) [reply src/components/common/stat/Stat.js]: 👍

File: test/components/common/stat/Stat.test.js (hunk 1 above)
Comment (cbwebdevelopment) [reply test/components/common/stat/Stat.test.js]: I filled in the test stub for assigning default input and suffix values from props, and deleted the test stub for calling onInputChange, which is actually handled in the `propagateInputChange` tests, and the `handleInputChange` and `handleSuffixChange` tests that assert that it's called.

File: test/components/common/stat/Stat.test.js (hunk 2 above)
Comment (cbwebdevelopment) [reply test/components/common/stat/Stat.test.js]: I've differentiated them.

**Comment by cbwebdevelopment** [comment]
@krystophv Ping for review of requested changes.

**Pull request merged**